### ISSUE TITLE
Can't use any protocols. Probably process handler issue

### ISSUE THREAD

**Issue by gr0ebi** [savonet/liquidsoap#691]
**Describe the bug**
None of the included protocols can be used (http/https/say...)

> main-run.log say test
2019/01/16 00:51:38 [server:3] New client: 127.0.0.1.
2019/01/16 00:51:45 [protocol.process:4] Processing wav,sox $(input) -c 2 $(output):text2wave:Test
2019/01/16 00:51:45 [protocol.process:4] Processing wav,echo "Test" | text2wave -scale 1.9 > $(output)
2019/01/16 00:51:45 [protocol.process:4] Executing echo "Test" | text2wave -scale 1.9 > "/tmp/liq-process029165.wav"
2019/01/16 00:51:45 [lang.run_process:4] Starting process
2019/01/16 00:51:45 [lang.run_process:4] Closing process's stdin
2019/01/16 00:52:04 [protocol.process:3] Failed to execute echo "Test" | text2wave -scale 1.9 > "/tmp/liq-process029165.wav": ("timeout","18.6649689674")
2019/01/16 00:52:04 [request:4] Failed to resolve "process:wav,echo \"Test\" | text2wave -scale 1.9 > $(output)"! For more info, see server command 'trace 5'.
2019/01/16 00:52:04 [protocol.process:3] Failed to resolve text2wave:Test
2019/01/16 00:52:04 [request:4] Failed to resolve "process:wav,sox $(input) -c 2 $(output):text2wave:Test"! For more info, see server command 'trace 4'.
2019/01/16 00:52:04 [lang.run_process:4] Cleaning up process
2019/01/16 00:52:15 [server:4] Timeout reached while communicating to client 127.0.0.1.
2019/01/16 00:52:15 [server:3] Client 127.0.0.1 disconnected.

> main-run.log http test
2019/01/16 01:04:25 [server:3] New client: 127.0.0.1.
2019/01/16 01:04:27 [procol.external:4] Fetching http head for http://www.hochmuth.com/mp3/Haydn_Cello_Concerto_D-1.mp3
2019/01/16 01:04:27 [protocol.process:4] Processing mp3,curl -sL "http$(colon)//www.hochmuth.com/mp3/Haydn_Cello_Concerto_D-1.mp3" -o $(output)
2019/01/16 01:04:27 [protocol.process:4] Executing curl -sL "http://www.hochmuth.com/mp3/Haydn_Cello_Concerto_D-1.mp3" -o "/tmp/liq-processfaad8d.mp3"
2019/01/16 01:04:27 [lang.run_process:4] Starting process
2019/01/16 01:04:27 [lang.run_process:4] Closing process's stdin
2019/01/16 01:04:47 [lang.run_process:4] Cleaning up process
2019/01/16 01:04:47 [protocol.process:3] Failed to execute curl -sL "http://www.hochmuth.com/mp3/Haydn_Cello_Concerto_D-1.mp3" -o "/tmp/liq-processfaad8d.mp3": ("timeout","19.4406847954")
2019/01/16 01:04:47 [request:4] Failed to resolve "process:mp3,curl -sL \"http$(colon)//www.hochmuth.com/mp3/Haydn_Cello_Concerto_D-1.mp3\" -o $(output)"! For more info, see server command 'trace 4'.
2019/01/16 01:04:57 [server:4] Timeout reached while communicating to client 127.0.0.1.
2019/01/16 01:04:57 [server:3] Client 127.0.0.1 disconnected.

**To Reproduce**

> #main.liq
set("server.telnet", true)
set("server.telnet.port", 1234)
set("harbor.bind_addr","0.0.0.0")
#set("harbor.verbose", true )
set("log.level", 4 )
#set("protocol.external.curl","curl")
#Music path
music1 = playlist("~/music/path/")
#Fallback
security = single("~/music/fail.mp3")
#Radio Config
radio = random([music1])
radio = fallback(track_sensitive = false,[smart_crossfade(radio),security])
#mic talkover test
#mic=input.harbor("mic",port=8085,password="dontpanic1764")
#request add test
mic = request.queue(id="mic")
radio = smooth_add(p=0.75, normal=radio, special=mic)
#Stream it out
output.icecast(%mp3(bitrate=192),
	host = "localhost", port = 8000,
	password = "hackme", mount = "192",
	name="Test", description="This is liquidsoap test music stream",radio)

> telnet 127.0.0.1 1234
mic.push say:Test

> telnet 127.0.0.1 1234
mic.push http://www.hochmuth.com/mp3/Haydn_Cello_Concerto_D-1.mp3

**Expected behavior**
Requested MP3 file and say command should be executed and smooth_add to current stream

**Version details**
 - OS: Debian buster
 - Version Linux radio01 4.9.0-3-amd64 #1 SMP Debian 4.9.30-2+deb9u5 (2017-09-19) x86_64 GNU/Linux

**Install method**
OPAM 2.0.2 via script (https://raw.githubusercontent.com/ocaml/opam/master/shell/install.sh)
OCaml 4.07.1
Liquidsoap 1.3.5
SoX v14.4.2
Festival Speech Synthesis System: 2.5.0:release December 2017
curl 7.52.1 (x86_64-pc-linux-gnu)

**Comment by gr0ebi** [comment]
just wanted to add that "/tmp/liq-process029165.wav" from "mic.push say:Test" is being created

**Comment by toots** [comment]
Thanks for reporting this issue. I've seen it before and will look at it asap.

**Comment by gr0ebi** [comment]
thanks a lot @toots

**Comment by gr0ebi** [comment]
Thanks for your reply
I upgraded liquidsoap with git but it's crashing now if I call any protocols:

> 2019/01/17 15:48:19 [server:3] New client: 127.0.0.1.
2019/01/17 15:48:21 [procol.external:4] Fetching http head for http://www.hochmuth.com/mp3/Haydn_Cello_Concerto_D-1.mp3
2019/01/17 15:48:21 [protocol.process:4] Processing mp3,curl -sL "http$(colon)//www.hochmuth.com/mp3/Haydn_Cello_Concerto_D-1.mp3" -o $(output)
2019/01/17 15:48:21 [protocol.process:4] Executing curl -sL "http://www.hochmuth.com/mp3/Haydn_Cello_Concerto_D-1.mp3" -o "/tmp/liq-processde5f20.mp3"
2019/01/17 15:48:21 [lang.run_process:4] Starting process
2019/01/17 15:48:21 [lang.run_process:4] Closing process's stdin
2019/01/17 15:48:22 >>> LOG START
2019/01/17 15:48:22 [main:3] Liquidsoap 1.3.5+scm (file:///home/webradio/liquidsoap@ec01a75f26a24939aa8895092ea8e492e76d5374:20190117:153303)
2019/01/17 15:48:22 [main:3] Using: bytes=[distributed with OCaml 4.02 or above] pcre=7.3.5 dtools=0.4.1 duppy=0.8.0 cry=0.6.2 mm=0.4.0 ogg=0.5.2 vorbis=0.7.1 opus=0.1.2 mad=0.4.5 flac=0.1.3 flac.ogg=0.1.3 dynlink=[distributed with Ocaml] lame=0.3.3 samplerate=0.1.4 taglib=0.3.3 magic=0.7.3 camomile=1.0.1
2019/01/17 15:48:22 [main:2]
2019/01/17 15:48:22 [main:2] DISCLAIMER: This version of Liquidsoap has been
2019/01/17 15:48:22 [main:2] compiled from a snapshot of the development code.
2019/01/17 15:48:22 [main:2] As such, it should not be used in production
2019/01/17 15:48:22 [main:2] unless you know what you are doing!

**Comment by toots** [comment]
Hmm. Yes indeed. There's a logic flaw, was too quick of a workaround heh. Let me rethink the implementation, thks!

**Comment by toots** [comment]
Ok, could you try now? Thks!

**Comment by gr0ebi** [comment]
still crashing, no changes in my logs

**Comment by toots** [comment]
Did you do:
```
cd ~/path/to/liquidsoap
git checkout workaround-691
git pull
opam reinstall liquidsoap
```
?

**Comment by gr0ebi** [comment]
forgot to reinstall :hushed:
It looks like its working now! thanks a lot @toots

**Comment by toots** [comment]
Awesome. Thanks. Now, let's clean this and merge it!

**Comment by toots** [comment]
This was fixed with merge of #703
